flink-autoscaler/src/main/java/org/apache/flink/autoscaler/ScalingExecutor.java

```diff
@@ -259,6 +220,9 @@ Map<JobVertexID, ScalingSummary> computeScalingSummary(
                                 if (parallelismChange.isNoChange()) {
                                     return;
                                 }
+                                if (parallelismChange.isOutsideUtilizationBound()) {
+                                    anyVertexOutsideBound.set(true);
+                                }
```

Comment:
We check the `OutsideUtilizationBound` of the final recommended parallelism instead of the `OutsideUtilizationBound` of latest recommended parallelism.

If the Utilization of all tasks is within range, we can skip scaling.